ipatests/test_integration/test_installation.py

```diff
@@ -479,3 +483,43 @@ def test_reserved_ip_as_forwarder(self):
         exp_str = ("Invalid IP Address 0.0.0.0: cannot use IANA reserved "
                    "IP address 0.0.0.0")
         assert exp_str in cmd.stdout_text
+
+
+class TestKRAinstallAfterCertRenew(IntegrationTest):
+    """ Test KRA installtion after ca agent cert renewal
+
+    KRA installtion was failing after ca-agent cert gets renewed.
+    This test check if the syptoms no longer exists.
+
+    related ticket: https://pagure.io/freeipa/issue/7288
+    """
+
+    def test_KRA_install_after_cert_renew(self):
+
+        tasks.install_master(self.master)
+
+        # get ca-agent cert contents
+        p12_content = self.master.get_file_contents(paths.DOGTAG_ADMIN_P12)
+
+        # load cert
+        cert = x509.load_der_x509_certificate(p12_content,
+                                              backend=default_backend())
+
+        # get expiry date of agent cert
+        cert_expiry = cert.not_valid_after()
+
+        # move date to grace period so that certs get renewed
+        grace_date = cert_expiry - timedelta(days=20)
+        grace_date = datetime.strftime(grace_date, "%Y-%m-%d %H:%M:%S")
+        self.master.run_command(['date', '-s', grace_date])
+
+        # wait sometime for cert renewal
+        sleep(200)
+
+        # move date after 3 days of actual expiry
+        cert_expiry = cert_expiry + timedelta(days=3)
+        cert_expiry = datetime.strftime(cert_expiry, "%Y-%m-%d %H:%M:%S")
+        self.master.run_command(['date', '-s', cert_expiry])
```

Comment:
I wonder what happens in Kerberos with all those time moves :thinking: Maybe something to look at should the test fail.

Reply:
Also, I just remembered that changing time in the RHEV lab VM machines might brick your access to the them (at least it was doing that back when I was playing with time-based rules). Keep that in mind (and possibly check that).